variables.tf

```diff
@@ -12,19 +12,13 @@ variable "name" {
   description = "The name of the Event Notifications instance that is created by this module."
 }
 
-variable "cos_destination_name" {
-  type        = string
-  description = "The name of the IBM Cloud Object Storage destination which will be created for the storage of failed delivery events."
-  default     = "COS Destination"
-}
-
 variable "cos_bucket_name" {
   type        = string
   description = "The name of an existing IBM Cloud Object Storage bucket which will be used for storage of failed delivery events. Required if `cos_integration_enabled` is set to true."
   default     = null
 }
 
-variable "cos_instance_id" {
+variable "existing_cos_instance_crn" {
```

Comment:
I'm not sure you need to change this variable name. The ID for COS is the CRN:
```
[
    {
        "guid": "46aed5b2-4c64-413f-a3c1-67338778b0fb",
        "id": "crn:v1:bluemix:public:cloud-object-storage:global:a/abac0df06b644a9cabc6e44f55b3880e:46aed5b2-4c64-413f-a3c1-67338778b0fb::",
        "url": "/v2/resource_instances/46aed5b2-4c64-413f-a3c1-67338778b0fb",
        "created_at": "2024-07-11T00:43:20.174505809Z",
        "updated_at": "2024-07-11T00:43:22.018459854Z",
        "deleted_at": null,
        "name": "def-ingres-wzfsae_cos",
        "region_id": "global",
        "account_id": "abac0df06b644a9cabc6e44f55b3880e",
        "resource_plan_id": "744bfc56-d12c-4866-88d5-dac9139e0e5d",
        "resource_group_id": "3501f4b799414cd7a0cd7d334278fba6",
        "crn": "crn:v1:bluemix:public:cloud-object-storage:global:a/abac0df06b644a9cabc6e44f55b3880e:46aed5b2-4c64-413f-a3c1-67338778b0fb::",
        "create_time": 0,
        "created_by": "IBMid-666000KAO3",
        "state": "active",
        "type": "service_instance",
        "resource_id": "dff97f5c-bc5e-4455-b470-411c3edbe49c",
        "dashboard_url": "https://cloud.ibm.com/objectstorage/crn%3Av1%3Abluemix%3Apublic%3Acloud-object-storage%3Aglobal%3Aa%2Fabac0df06b644a9cabc6e44f55b3880e%3A46aed5b2-4c64-413f-a3c1-67338778b0fb%3A%3A",
        "allow_cleanup": false,
        "locked": false,
        "last_operation": {
            "type": "create",
            "state": "succeeded",
            "description": "Completed create instance operation",
            "updated_at": null,
            "cancelable": false
        },
        "account_url": "",
        "resource_plan_url": "",
        "resource_bindings_url": "/v2/resource_instances/46aed5b2-4c64-413f-a3c1-67338778b0fb/resource_bindings",
        "resource_aliases_url": "/v2/resource_instances/46aed5b2-4c64-413f-a3c1-67338778b0fb/resource_aliases",
        "siblings_url": "",
        "target_crn": "crn:v1:bluemix:public:globalcatalog::::deployment:744bfc56-d12c-4866-88d5-dac9139e0e5d%3Aglobal"
    }
]
```

Reply:
> I'm not sure you need to change this variable name. The ID for COS is the CRN:
> 
> ```
> [
>     {
>         "guid": "46aed5b2-4c64-413f-a3c1-67338778b0fb",
>         "id": "crn:v1:bluemix:public:cloud-object-storage:global:a/abac0df06b644a9cabc6e44f55b3880e:46aed5b2-4c64-413f-a3c1-67338778b0fb::",
>         "url": "/v2/resource_instances/46aed5b2-4c64-413f-a3c1-67338778b0fb",
>         "created_at": "2024-07-11T00:43:20.174505809Z",
>         "updated_at": "2024-07-11T00:43:22.018459854Z",
>         "deleted_at": null,
>         "name": "def-ingres-wzfsae_cos",
>         "region_id": "global",
>         "account_id": "abac0df06b644a9cabc6e44f55b3880e",
>         "resource_plan_id": "744bfc56-d12c-4866-88d5-dac9139e0e5d",
>         "resource_group_id": "3501f4b799414cd7a0cd7d334278fba6",
>         "crn": "crn:v1:bluemix:public:cloud-object-storage:global:a/abac0df06b644a9cabc6e44f55b3880e:46aed5b2-4c64-413f-a3c1-67338778b0fb::",
>         "create_time": 0,
>         "created_by": "IBMid-666000KAO3",
>         "state": "active",
>         "type": "service_instance",
>         "resource_id": "dff97f5c-bc5e-4455-b470-411c3edbe49c",
>         "dashboard_url": "https://cloud.ibm.com/objectstorage/crn%3Av1%3Abluemix%3Apublic%3Acloud-object-storage%3Aglobal%3Aa%2Fabac0df06b644a9cabc6e44f55b3880e%3A46aed5b2-4c64-413f-a3c1-67338778b0fb%3A%3A",
>         "allow_cleanup": false,
>         "locked": false,
>         "last_operation": {
>             "type": "create",
>             "state": "succeeded",
>             "description": "Completed create instance operation",
>             "updated_at": null,
>             "cancelable": false
>         },
>         "account_url": "",
>         "resource_plan_url": "",
>         "resource_bindings_url": "/v2/resource_instances/46aed5b2-4c64-413f-a3c1-67338778b0fb/resource_bindings",
>         "resource_aliases_url": "/v2/resource_instances/46aed5b2-4c64-413f-a3c1-67338778b0fb/resource_aliases",
>         "siblings_url": "",
>         "target_crn": "crn:v1:bluemix:public:globalcatalog::::deployment:744bfc56-d12c-4866-88d5-dac9139e0e5d%3Aglobal"
>     }
> ]
> ```

@ocofaigh I see both id and crn are same.
```
 "id": "crn:v1:bluemix:public:cloud-object-storage:global:a/abac0df06b644a9cabc6e44f55b3880e:46aed5b2-4c64-413f-a3c1-67338778b0fb::",

"crn": "crn:v1:bluemix:public:cloud-object-storage:global:a/abac0df06b644a9cabc6e44f55b3880e:46aed5b2-4c64-413f-a3c1-67338778b0fb::",
```
Should I change the variable name to `existing_cos_instance_id` instead of `existing_cos_instance_crn`?

Reply:
lets just leave it as is so its minimal disruption for consumers

Reply:
@Ak-sky are you not changing this back? Trying to reduce the impact on consumers here, and I see no reason to change the variable name here